compiler/optimizer/IsolatedStoreElimination.cpp

```diff
@@ -202,12 +202,7 @@ int32_t TR_IsolatedStoreElimination::perform()
    int32_t cost = 0;
 
    // The non-useDefInfo path examines direct loads/stores to autos/parms to determine what
-   // stores are removable.  Autos can have their address taken in WCode and this aliasing
-   // is only required to be communicated through PALIs (pointer aliasing) on consuming calls.
-   // Therefore aliasing (through useDefInfo) must be consulted to safely remove stores.
-   // Checking address taken is not sufficient as a frontend does not have to set address taken
-   // if the aliasing is correctly communicated through PALIs (address taken pessimizes optimizing
-   // the symbol everywhere but PALIs can provide more precise information)
+   // stores are removable.
```

Comment:
I believe there's an opportunity to do a bit more clean up here relating to this code.  I'd rather have this completed now as part of this PR because once this comment is gone it won't draw as much attention.

It seems to me that there are now only two cases in the conditional below:  1) when `useDefInfo` is non-NULL, and 2) when `useDefInfo` is NULL.  My inspection of the `if` statement below reveals that it could be simplified into just:
```
   if (useDefInfo)
      {
      ... with use def info
      }
   else
      {
      ... without use def info
      }
```
i.e., delete the 3rd branch of the original `if`.

Also, this comment that you modified should be moved to the `else` branch as that's what it applies to.

@andrewcraik for his awareness.